src/main/kotlin/lotto/domain/LottoProvider.kt

```diff
@@ -1,10 +1,34 @@
 package lotto.domain
 
-class LottoProvider(payment: Int) {
-    val numberOfLottos: Int = payment / LOTTO_PRICE
-    val lottos: List<Lotto> = (1..numberOfLottos).map { Lotto() }
+import lotto.domain.numbers.LottoNumbers
+
+class LottoProvider(payment: Int, customLottoNumbers: List<LottoNumbers> = emptyList()) {
+    val numberOfCustomLottos: Int
+    val numberOfAutomaticLottos: Int
+
+    val lottos: List<Lotto>
+
+    init {
+        numberOfCustomLottos = when (isAffordable(payment, customLottoNumbers.size)) {
+            true -> customLottoNumbers.size
+            false -> throw IllegalArgumentException(TOO_MANY_LOTTO_REQUESTED)
+        }
+
+        numberOfAutomaticLottos = payment.getNumberOfAffordableLotto() - numberOfCustomLottos
+
+        lottos = generateCustomLottos(customLottoNumbers) + generateAutomaticLottos(numberOfAutomaticLottos)
+    }
+
+    private fun generateCustomLottos(customLottoNumbers: List<LottoNumbers>) = customLottoNumbers.map { Lotto(it) }
```

Comment:
코틀린 확장함수를 사용하면 조금더 읽기 수월할수 있을것 같습니다~

```suggestion
        lottos = customLottoNumbers.generateCustomLottos() + generateAutomaticLottos(numberOfAutomaticLottos)
    }

    private fun List<LottoNumbers>.generateCustomLottos() = this.map { Lotto(it) }
```

Reply:
[f5dd586](https://github.com/next-step/kotlin-lotto/pull/386/commits/f5dd586f03ae9f8b9ede95b99064b32d8116cbfd) 에 반영했습니다!